src/include/migraphx/common_header.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_COMMON_HEADER_HPP
+#define MIGRAPHX_GUARD_RTGLIB_COMMON_HEADER_HPP
+#include <migraphx/program.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/operators.hpp>
+#include <migraphx/iterator_for.hpp>
+#include <migraphx/pass_config.hpp>
+#include <migraphx/config.hpp>
+
+#include <set>
+#include <list>
+#include <vector>
+#include <queue>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+//#define MIGRAPHX_DEBUG_OPT
+
+#ifdef MIGRAPHX_DEBUG_OPT
+#define MIGRAPHX_DEBUG(s) s
+#else
+#define MIGRAPHX_DEBUG(s)
+#endif // MIGRAPHX_DEBUG_OPT
+} // namespace MIGRAPHX_INLINE_NS
+} // namespace migraphx
+
+#endif // MIGRAPHX_GUARD_RTGLIB_COMMON_HEADER_HPP
```

Comment:
Common headers just slow compilation. Instead move `MIGRAPHX_DEBUG_OPT` to a debug header, and include the rest of the headers where they are needed.

Reply:
quite contrary,  using common headers plus macro definitions makes common headers only be included once.   This helps reduce compilation time and is a common practice we do in the compiler.  Anyway,

Reply:
If its only included once then why is a separate header needed? 

Usually common headers are included in multiple places but they dont need everything included in the common headers. For example, a file might need the `MIGRAPHX_DEBUG_OPT` but doesnt need vector or queue, which results in unnecessary compilation of extra headers.

Reply:
Yes, it was meant to be used to be included more than once.   It is used only once now.  Anyway, I will remove common header.

Reply:
fixed